scripts/convex.tenderly.ts

```diff
@@ -28,12 +28,16 @@ async function main(): Promise<void> {
       gas_price: '0',
       value: '0',
       save: false,
-      simulation_type: 'quick', // 'full' uses contract source code for a more detailed trace and is slower
+      use_cache: true,
```

Comment:
Just tested out the changes on the branch, here are the results:
- The first run failed with a 400 error—this always happens to me the first time I run this tenderly sim after not having run it for a while. Perhaps too much data to load so it times out?
- The second run (first successful run) took 0m6.359s
- Then did 10 more runs which had a median of 0m1.9315s 
- Then set `use_cache` to false which gave a 10 run median of 0m3.072s

I'm curious to know what explains these various differences. My guess is the first run may have timed out, the 1.9s vs. 3.1s vs 6.4s are related to various caching layers? But when would you want `use_cache` to be true vs. false? Would love any details you can share!

Reply:
Yeah of course! I'd like to address the 400 issue though I have no idea why it happens (I've never had it happen), it's most likely not due to the size/complexity of the calculation, do you get a response immediately or is it a delayed 400?

As for the layers of caching, there is a global lru cache for accessed state for simulations/transactions and other more implicit caching layers there. The flag itself is on our side and represents a more aggressive caching. You’d never want to have it on false, unless you’re explicitly testing cases where you want to avoid cache hits (i.e when debugging something).

Also for some further clarification, are we benchmarking the time reported by the `time` command in bash, or the actual duration of the simulation (the time reported by console.time)?

Reply:
>  I'd like to address the 400 issue though I have no idea why it happens (I've never had it happen), it's most likely not due to the size/complexity of the calculation, do you get a response immediately or is it a delayed 400?

It's a delayed 400. I changed the block number to 13445961 and got the 400 again. If you aren't able to reproduce, perhaps try an earlier block number than that. First I tried the most recent block number and didn't get a 400, so then tried an earlier block. This aligns with my suspicion that it's related to caching/availability of data and maybe timing out as a result (and does seem to be consistent with the fact that there's an lru cache)

> As for the layers of caching, there is a global lru cache for accessed state for simulations/transactions and other more implicit caching layers there. The flag itself is on our side and represents a more aggressive caching. You’d never want to have it on false, unless you’re explicitly testing cases where you want to avoid cache hits (i.e when debugging something).

Got it, thanks! Perhaps worth making `use_cache: true` the default when it's not specified then? That way users can get the extra speed without having to making sure the enable the cache.

> Also for some further clarification, are we benchmarking the time reported by the time command in bash, or the actual duration of the simulation (the time reported by console.time)?

Currently we use the time reported by the `time` command. I definitely see the argument for reporting the true simulation duration, though. The reason I think `time` command is the way to go for now is:
- It represents the actual time a user would experience, which is ultimately what matters in my opinion
- Foundry and dapptools don't currently expose a way to extract pure sim time

FWIW I do want to add another simulation to this repo that uses a ton of gas but doesn't rely on mainnet data/state as a second comparison

Reply:
> Got it, thanks! Perhaps worth making use_cache: true the default when it's not specified then? That way users can get the extra speed without having to making sure the enable the cache.

Right, and we'll make it enabled by default, it's just currently new and we decided it would be best to have it be opt-in.

> It represents the actual time a user would experience, which is ultimately what matters in my opinion
Well it introduces the script overhead (which will most likely be higher in real world examples), but I see your point.

>Foundry and dapptools don't currently expose a way to extract pure sim time
Not that familiar with these two, however they seem to be invoked directly for the simulation, rather than having the script overhead. I'm assuming there is no different way to invoke those, where as for js scripts there are many different ways.

One more thing running through my head is that it might be really cool to add an action that runs the benchmark (either on update/manual/periodic). How does that sound?

Reply:
Just updated the benchmark in the README, will merge unless there's anything else left to do on this branch (I don't think there is, just confirming)

Re github actions, agree having CI run the benchmarks and publish results somewhere would be great. Tracking that in https://github.com/mds1/convex-shutdown-simulation/issues/6, definitely open to PRs there if you wanted to take the lead since I haven't gotten around to it yet 🙂

Reply:
Yeah, nothing else to do here. Feel free to merge 😅